homeassistant/components/syncthing/__init__.py

```diff
@@ -0,0 +1,163 @@
+"""The syncthing integration."""
+import asyncio
+import logging
+
+import aiosyncthing
+import voluptuous as vol
+
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import (
+    CONF_NAME,
+    CONF_TOKEN,
+    CONF_URL,
+    CONF_VERIFY_SSL,
+    EVENT_HOMEASSISTANT_STOP,
+)
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.dispatcher import async_dispatcher_send
+
+from .const import (
+    DOMAIN,
+    EVENTS,
+    RECONNECT_INTERVAL,
+    SERVER_AVAILABLE,
+    SERVER_UNAVAILABLE,
+)
+
+CONFIG_SCHEMA = vol.Schema({DOMAIN: vol.Schema({})}, extra=vol.ALLOW_EXTRA)
+
+PLATFORMS = ["sensor"]
+
+_LOGGER = logging.getLogger(__name__)
+
+
+async def async_setup(hass: HomeAssistant, config: dict):
+    """Set up the syncthing component."""
+    return True
+
+
+async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry):
+    """Set up syncthing from a config entry."""
+    data = entry.data
+
+    if DOMAIN not in hass.data:
+        hass.data[DOMAIN] = {}
+
+    name = data[CONF_NAME]
+
+    client = aiosyncthing.Syncthing(
+        data[CONF_TOKEN],
+        url=data[CONF_URL],
+        verify_ssl=data[CONF_VERIFY_SSL],
+        loop=hass.loop,
+    )
+
+    syncthing = SyncthingClient(hass, client, name)
```

Comment:
This particular line cannot fail. I raise `ConfigEntryNotReady` from [sensor.py](https://github.com/home-assistant/core/pull/38331/files#diff-78670fbe56de345126112773448b32b6R54) in case if the server is not available.

If you think it makes sense to raise it from here, I can add an extra check

Reply:
Instead of raising it from the specific sensor (for each sensor again), this can be handled more centrally. When a ConfigEntryNotReady is raised, Home Assistant will automatically retry setting up the entry later again.

Reply:
I raise it once from function `async_setup_entry` in `sensor.py` and home assistant automatically retries setting up the entry

Reply:
In fact if I raise `ConfigEntryNotReady` from this method hass does not catch it and print a stack trace to the logs.